# HG changeset patch
# User Rik <rik@octave.org>
# Date 1512155306 28800
#      Fri Dec 01 11:08:26 2017 -0800
# Node ID 1b17a5d82e3f69910ef80936da20f9cd7c1b26b7
# Parent  30bbef7bf9197e568b592234d8389516777d1b34
eigs.m: Fix failing BIST test (bug #49523).

* eigs.m: Use isreal() to determine whether NaN or NaN+1i*NaN should be the
expected value.

diff --git a/scripts/sparse/eigs.m b/scripts/sparse/eigs.m
--- a/scripts/sparse/eigs.m
+++ b/scripts/sparse/eigs.m
@@ -1382,18 +1382,21 @@ endfunction
 %!testif HAVE_ARPACK
 %! A = toeplitz ([0, 1, zeros(1, 8)], [0, -1, zeros(1, 8)]);
 %! A(1, 1) = 1;
 %! A = kron (A, eye (10)) + kron (eye (10), A);
 %! opts.v0 = (1:100)';
 %! opts.maxit = 1;
 %! warning ("off", "Octave:eigs:UnconvergedEigenvalues", "local");
 %! d = eigs (A, 4, "sm", opts);
-%! assert (d(4), NaN);
-%! assert (imag (d(4)), 0);
+%! if (isreal (d))
+%!   assert (d(4), NaN);
+%! else
+%!   assert (d(4), NaN +1i*NaN);
+%! endif
 %!testif HAVE_ARPACK
 %! A = magic (100) / 10 + eye (100);
 %! opts.v0 = (1:100)';
 %! opts.maxit = 10;
 %! warning ("off", "Octave:eigs:UnconvergedEigenvalues", "local");
 %! d = eigs (A, 10, "sm", opts);
 %! assert (d(10), NaN+1i*NaN);
 %!testif HAVE_ARPACK
